scheduling-api-graphql/scheduling-api-graphql-fetchers/src/main/java/org/ow2/proactive/scheduling/api/graphql/fetchers/converter/JobInputConverter.java

```diff
@@ -89,6 +95,16 @@
             String priority = i.getPriority();
             String projectName = i.getProjectName();
             String status = i.getStatus();
+            VariablesInput variablesInput = i.getWithVariables();
+
+            if (variablesInput != null) {
+                for (Map<String, String> o : variablesInput.getVariables()) {
+                    MapJoin<JobData, String, JobDataVariable> joinMap = root.joinMap("variables", JoinType.LEFT);
```

Comment:
shouldn't the join be done once and not for each variable ?

Reply:
I'm afraid we do need to join for each of the key/value pair. In our case rows are the result of the OR operation so we can only constraint the AND on columns. That's why I'm also using the DISTINCT operator (or I would have duplicated jobs that satisfy more than one predicate).